lib/std/os/linux/s390x.zig

```diff
@@ -0,0 +1,281 @@
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+        : "memory"
+    );
+}
+
+pub fn syscall1(number: SYS, arg1: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+        : "memory"
+    );
+}
+
+pub fn syscall2(number: SYS, arg1: usize, arg2: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+        : "memory"
+    );
+}
+
+pub fn syscall3(number: SYS, arg1: usize, arg2: usize, arg3: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+        : "memory"
+    );
+}
+
+pub fn syscall4(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+          [arg4] "{r5}" (arg4),
+        : "memory"
+    );
+}
+
+pub fn syscall5(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+          [arg4] "{r5}" (arg4),
+          [arg5] "{r6}" (arg5),
+        : "memory"
+    );
+}
+
+pub fn syscall6(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize, arg6: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+          [arg4] "{r5}" (arg4),
+          [arg5] "{r6}" (arg5),
+          [arg6] "{r7}" (arg6),
+        : "memory"
+    );
+}
+
+pub fn clone() callconv(.Naked) usize {
+    asm volatile (
+        \\.text
+        \\.global __clone
+        \\.hidden __clone
+        \\.type __clone, %function
+        \\__clone:
+        \\  # int clone(
+        \\  #    fn,      a = r2
+        \\  #    stack,   b = r3
+        \\  #    flags,   c = r4
+        \\  #    arg,     d = r5
+        \\  #    ptid,    e = r6
+        \\  #    tls,     f = *(r15+160)
+        \\  #    ctid)    g = *(r15+168)
+        \\  #
+        \\  # pseudo C code:
+        \\  # tid = syscall(SYS_clone,b,c,e,g,f);
+        \\  # if (!tid) syscall(SYS_exit, a(d));
+        \\  # return tid;
+        \\
+        \\  # preserve call-saved register used as syscall arg
+        \\  stg  %r6, 48(%r15)
```

Comment:
```suggestion
        \\  stg  %%r6, 48(%%r15)
```

And likewise throughout.

Reply:
what difference does this make? it doesnt change the results of the behavior or std tests

Reply:
You need to use double `%%` to obtain a literal `%` since [the latter is used as the escape character when referring to inputs/outputs](https://ziglang.org/documentation/master/#Assembly).

Are you sure you're running any tests that actually end up calling this? This code is only used when glibc/musl aren't linked.

Reply:
gotcha